reactor-core/src/main/java/reactor/core/publisher/Operators.java

```diff
@@ -618,6 +618,30 @@ public static void onDiscardMultiple(@Nullable Iterator<?> multiple, boolean kno
 		}
 	}
 
+	public static void onDiscardMultiple(@Nullable Spliterator<?> multiple, boolean knownToBeFinite, Context context) {
+		if (multiple == null) return;
+		if (!knownToBeFinite) return;
+
+		Consumer<Object> hook = context.getOrDefault(Hooks.KEY_ON_DISCARD, null);
+		if (hook != null) {
+			try {
+				multiple.forEachRemaining(o -> {
+					if (o != null) {
+						try {
+							hook.accept(o);
+						}
+						catch (Throwable t) {
+							log.warn("Error while discarding element from an Iterator, continuing with next element", t);
+						}
+					}
+				});
+			}
+			catch (Throwable t) {
+				log.warn("Error while discarding Iterator, stopping", t);
```

Comment:
Consider Splitterator to make debugging easier.

Reply:
fixed